src/drivers/gitea_test.go

```diff
@@ -0,0 +1,226 @@
+package drivers_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/git-town/git-town/src/drivers"
+	"github.com/stretchr/testify/assert"
+	httpmock "gopkg.in/jarcoal/httpmock.v1"
+)
+
+type mockGiteaConfig struct {
+	codeHostingDriverName string
+	remoteOriginURL       string
+	giteaToken            string
+	configuredHostName    string
+}
+
+type mockGiteaEndpoints struct {
+	root     string
+	orga     string
+	repo     string
+	version  string
+	prBase   string
+	prOpen   string
+	pr1      string
+	pr1Merge string
+}
+
+func newMockGiteaEndpoints() (mge mockGiteaEndpoints) {
+	mge = mockGiteaEndpoints{}
+	mge.root = "https://gitea.com/api/v1"
+	mge.orga = "git-town"
+	mge.repo = "git-town"
+	mge.version = mge.root + "/version"
+	mge.prBase = mge.root + "/repos/" + mge.orga + "/" + mge.repo + "/pulls"
+	mge.prOpen = mge.prBase + "?limit=50&page=0&state=open"
+	mge.pr1 = mge.prBase + "/1"
+	mge.pr1Merge = mge.pr1 + "/merge"
+	return mge
+}
+
+func (mgc mockGiteaConfig) GetCodeHostingDriverName() string {
```

Comment:
Move these method definitions to line 19 to keep them together with the definition of the struct?